File: marketmotormicroservices/producto-service/src/main/java/com/jma/productoservice/detalleOrdenCompra/domain/response/DetalleOrdenCompraResponse.java
package com.jma.productoservice.detalleOrdenCompra.domain.response;

import com.jma.productoservice.detalleOrdenCompra.domain.dto.DetalleOrdenCompraDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class DetalleOrdenCompraResponse {

    private List<DetalleOrdenCompraDto> content;
    private int pageNo;
    private int pageSize;
    private long totalElements;
    private int totalPages;
    private boolean last;

}

File: frontend-marketmotor/src/app/models/responseapi/DetalleOrdenCompraResponse.ts
import { DetalleOrdenCompra } from "../dtos/DetalleOrdenCompra";

export class DetalleOrdenCompraResponse{

    content: DetalleOrdenCompra[];
    pageNo: number;
    pageSize: number;
    totalElements: number;
    totalPages: number;
    last: boolean;

    constructor(){
        this.content  = [new DetalleOrdenCompra()]
        this.pageNo = 0;
        this.pageSize = 0;
        this.totalElements = 0;
        this.totalPages = 0;
        this.last = false;
    }

}